experimental/cmd/mfrc522/main.go

```diff
@@ -0,0 +1,144 @@
+// Copyright 2018 The Periph Authors. All rights reserved.
+// Use of this source code is governed under the Apache License, Version 2.0
+// that can be found in the LICENSE file.
+
+package main
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"strconv"
+	"strings"
+
+	"periph.io/x/periph/conn/gpio/gpioreg"
+	"periph.io/x/periph/conn/spi/spireg"
+	"periph.io/x/periph/experimental/devices/mfrc522"
+	"periph.io/x/periph/experimental/devices/mfrc522/commands"
+	"periph.io/x/periph/host"
+)
+
+func mainImpl() error {
+
+	sector := flag.Int("sector", 1, "Sector to access")
+	block := flag.Int("block", 0, "Block to access")
+
+	rsPin := flag.String("rs-pin", "", "Reset pin")
+	irqPin := flag.String("irq-pin", "", "IRQ pin")
+
+	keyCommand := flag.Bool("wa", false, "Overwrite keys")
+	blockCommand := flag.String("wb", "", "Overwrite block by provided data (comma-separated list of 16 bytes)")
+
+	spiID := flag.String("spi", "", "SPI device")
+
+	key := flag.String("key", "", "Comma-separated key bytes")
+
+	flag.Parse()
+
+	if *irqPin == "" || *rsPin == "" {
+		return errors.New("please provide -rs-pin and -irq-pin arguments, or -h for help")
+	}
+
+	if _, err := host.Init(); err != nil {
+		return err
+	}
+
+	currentAccessKey := mfrc522.DefaultKey
+
+	if *key != "" {
+		keyBytes := strings.SplitN(*key, ",", 6)
+		if len(keyBytes) != 6 {
+			return errors.New("key should consist of 6 decimal numbers")
+		}
+		currentAccessKey = make([]byte, 6)
+
+		for i, v := range keyBytes {
+			intV, err := strconv.ParseUint(v, 10, 8)
+			if err != nil {
+				return err
+			}
+			currentAccessKey[i] = byte(intV)
+		}
+	}
+
+	currentAccessMethod := byte(commands.PICC_AUTHENT1B)
+
+	spiDev, err := spireg.Open(*spiID)
+	if err != nil {
+		return err
+	}
+
+	rfid, err := mfrc522.NewSPI(spiDev, gpioreg.ByName(*rsPin), gpioreg.ByName(*irqPin))
```

Comment:
In practice, you should confirm that ByName() call didn't return nil, as you need to inform the user that the pin name doesn't exist.

Reply:
I think that it could be wise to check that the passed pin references are not nil inside the `NewSPI` method too, to make it a bit more defensive.

Reply:
No driver does that. We can't start checking for nil for the pins but not the spi bus as it would be inconsistent. Then we'd have to start checking for nil everywhere on every driver. I prefer the default that passing nil is never supported.